osgp/platform/osgp-secret-management/src/main/java/org/opensmartgridplatform/secretmanagement/application/endpoints/SecretManagementEndpoint.java

```diff
@@ -0,0 +1,99 @@
+
+import java.util.List;
+
+import lombok.extern.slf4j.Slf4j;
+import org.opensmartgridplatform.ws.schema.core.secret.management.GetSecretsRequest;
+import org.opensmartgridplatform.ws.schema.core.secret.management.GetSecretsResponse;
+import org.opensmartgridplatform.ws.schema.core.secret.management.OsgpResultType;
+import org.opensmartgridplatform.ws.schema.core.secret.management.StoreSecretsRequest;
+import org.opensmartgridplatform.ws.schema.core.secret.management.StoreSecretsResponse;
+import org.opensmartgridplatform.ws.schema.core.secret.management.TypedSecrets;
+import org.opensmartgridplatform.secretmanagement.application.domain.SecretType;
+import org.opensmartgridplatform.secretmanagement.application.domain.TypedSecret;
+import org.opensmartgridplatform.secretmanagement.application.services.SecretManagementService;
+import org.opensmartgridplatform.shared.exceptionhandling.ComponentType;
+import org.opensmartgridplatform.shared.exceptionhandling.OsgpException;
+import org.opensmartgridplatform.shared.exceptionhandling.TechnicalException;
+import org.springframework.ws.server.endpoint.annotation.Endpoint;
+import org.springframework.ws.server.endpoint.annotation.PayloadRoot;
+import org.springframework.ws.server.endpoint.annotation.RequestPayload;
+import org.springframework.ws.server.endpoint.annotation.ResponsePayload;
+
+@Endpoint
+@Slf4j
+public class SecretManagementEndpoint {
+
+    private static final String NAMESPACE_URI = "http://www.opensmartgridplatform"
+            + ".org/schemas/security/secretmanagement/2020/05";
+
+    private final SecretManagementService secretManagementService;
+    private final SoapEndpointDataTypeConverter converter;
+
+    public SecretManagementEndpoint(SecretManagementService secretManagementService,
+            SoapEndpointDataTypeConverter converter) {
+        this.secretManagementService = secretManagementService;
+        this.converter = converter;
+    }
+
+    @PayloadRoot(namespace = NAMESPACE_URI, localPart = "getSecretsRequest")
+    @ResponsePayload
+    public GetSecretsResponse getSecretsRequest(@RequestPayload GetSecretsRequest request) throws OsgpException {
+
+        log.info("Handling incoming SOAP request 'getSecretsRequest' for device {}", request.getDeviceId());
+
+        log.trace(request.toString());
+
+        try {
+            GetSecretsResponse response = new GetSecretsResponse();
+
+            List<SecretType> secretTypeList = converter.convertToSecretTypes(request.getSecretTypes());
+            List<TypedSecret> typedSecrets = secretManagementService.retrieveSecrets(request.getDeviceId(),
+                    secretTypeList);
+
+            TypedSecrets soapTypedSecrets = converter.convertToSoapTypedSecrets(typedSecrets);
+
+            response.setTypedSecrets(soapTypedSecrets);
+            response.setResult(OsgpResultType.OK);
+
+            log.trace(response.toString());
+
+            return response;
+        } catch (Exception e) {
```

Comment:
Any exception will be translated to a TechnicalException. Is this intended?
Maybe return a SOAP Fault to communicate the problem to the caller?